File: DAM1/T6_Herencia_Trabajadores/src/model/Asalariado.java
package model;

public class Asalariado extends Trabajador{

    private double sueldo;
    private int numeroPagas;
    private boolean contratodo;

    public Asalariado() {}

    public Asalariado(String nombre, String apellido, String dni, double sueldo, int numeroPagas, boolean contratodo) {
        super(nombre, apellido, dni);
        this.sueldo = sueldo;
        this.numeroPagas = numeroPagas;
        this.contratodo = contratodo;
    }

    public void mostrarDatos() {
        super.mostrarDatos();
        System.out.println("Salario Anual: "+(sueldo * numeroPagas));
        System.out.println("Salario Mensual: "+sueldo);
        System.out.println("Numero de Pagas: "+numeroPagas);
    }

    public double getSueldo() {
        return sueldo;
    }

    public void setSueldo(double sueldo) {
        this.sueldo = sueldo;
    }

    public int getNumeroPagas() {
        return numeroPagas;
    }

    public void setNumeroPagas(int numeroPagas) {
        this.numeroPagas = numeroPagas;
    }

    public boolean isContratodo() {
        return contratodo;
    }

    public void setContratodo(boolean contratodo) {
        this.contratodo = contratodo;
    }
}

File: DAM2_Pro/PMDM/Tema_1/TrabajoDeHerencia/src/model/Asalariado.kt
package model

import kotlin.random.Random

class Asalariado(nombre: String, apellido: String, dni: String, salario: Double, var nPagas: Int, var irof: Double)
    : Trabajador(nombre,apellido,dni,salario){


    fun pedirAumento() {
        var numRandom = Random.nextInt(1,11);
       if (numRandom < 5) {
           println("Aumentar no, loshara")
       } else {
           salario *= 1.1;
           println("Salario aumentado a 10 porciento")
       }
    }


    override fun calcularSalarioNeto(): Double {
        return salario - (salario * irof)
    }

    override fun mostrarDatos() {
        super.mostrarDatos()
        println("nPagas = ${nPagas}")
        println("irof = ${irof}")
    }
}